packages/editor/src/components/post-featured-image/index.js

```diff
@@ -49,6 +49,7 @@ function PostFeaturedImage( { currentPostId, featuredImageId, onUpdateImage, onR
 					<MediaUpload
 						title={ postLabel.featured_image || DEFAULT_FEATURE_IMAGE_LABEL }
 						onSelect={ onUpdateImage }
+						featuredImage
```

Comment:
I'd like to understand why is this different from regular media modals? It doesn't seem like a good prop as it's an obscure prop which we don't really know what it does behind the scenes. Can this be something like `showAuthorFilters` or things like that? Why is this behavior specific to feature images?

Reply:
The regular media modal uses a default frame for selecting media. In gallery we use a `gallery` prop in the same way as this PR that gives `MediaUpload` an indicator of which frame to use. I think of this prop as sort of a context to opening the modal. If gallery, open gallery frame. If featured image, open featured image frame. This can get messy as we add more conditions (classic-gallery will also pass a prop similar)

I used `featuredImage` to select the following frame, which is not so much specific to showing author filters, but rather the featured image frame that we have used in WordPress historically:

https://github.com/WordPress/gutenberg/blob/684f2260a9731fa2f5dc1e3ced918fbf1195d13e/packages/edit-post/src/hooks/components/media-upload/index.js#L12-L26

@joemcgill do you have any thoughts on how to do this differently?

Reply:
I don't really care about how we implement this internally, I'm just trying to understand what's different between these frames and how to clarify it in the external API of the component. a `frame` is not something the consumer of the component understands.

Reply:
I went ahead and renamed it. I should warn though that adding `showAuthorFilters` to MediaUpload will not make the author filters show for other media flows. It will show all of the attributes of the featured image frame instead.

For instance, adding it to the image block's implementation of MediaUpload (the edit button).
```
						<MediaUpload
							onSelect={ this.onSelectImage }
							allowedTypes={ ALLOWED_MEDIA_TYPES }
							showAuthorFilters
							value={ id }
							render={ ( { open } ) => (
								<IconButton
									className="components-toolbar__control"
									label={ __( 'Edit image' ) }
									icon="edit"
									onClick={ open }
								/>
							) }
						/>
```

forces that button to open the featured image flow of selecting media. This sets the featured image button to "Set featured image" which is wrong in this context.

Reply:
@antpb Gotcha, so there's no way in the media library to distinguish the "featured image" flow from the fact that we want author filters?

Reply:
>  there's no way in the media library to distinguish the "featured image" flow from the fact that we want author filters?

Thanks for the updates @antpb and in that case, I'd agree we should try to find a "compromise". So I'd be ok with a `featuredImage` prop if it's explicitely marked as unstable, maybe at some point the media library could become more "pluggable" and we'd change it.

What about naming the prop `unstableFeaturedImageFlow` or something like that? It clarifies that this is not something third-party code should rely on as it can get removed/changed at any time?

Reply:
We should also address the `gallery` prop here because that also initiates the gallery flow and follows the same approach I took in this PR.

Reply:
Agreed it's kind of similar. My initial reaction for the `gallery` was to think that it was different because it can also mean `multiple` but the thing is that we already have a `multiple` prop :)

Reply:
Love that compromise @youknowriad !  Do you think `gallery` should be renamed the same?

Reply:
@antpb Seems like it should but in that case we should think at keeping it working for a version or two (`deprecated` call)

Reply:
@youknowriad:

> I'd like to understand why is this different from regular media modals? It doesn't seem like a good prop as it's an obscure prop which we don't really know what it does behind the scenes.

There are a few things happening here. First, in the classic editor, there are distinct media modal instances (i.e., frames) created when you are inserting media in post content versus when you are changing the featured image from the Featured Image post meta box (there are other modal instances available in WP, but that's outside the scope of this discussion). 

Each media modal that is initialized can include a set of "states" for controlling a specific stage in the flow required by that modal (i.e., selecting, editing, inserting, etc). Each state is represented as a Backbone controller ([found here](https://github.com/WordPress/wordpress-develop/tree/master/src/js/media/controllers)), and are loaded into individual modal instances when that particular media "frame". 

The default modal in the classic editor includes all states necessary to support any media flows in the editor—including even (surprisingly) selecting the featured image of the current post from the "Add Media" button. You can see  here how [the default post frame loads all of the different state controllers](https://github.com/WordPress/wordpress-develop/blob/master/src/js/media/views/frame/post.js#L48-L174)).

The challenge with the `MediaUpload` component (BTW, is it too late to rename this, because `MediaLibrary` would be much more clear) is that its purpose is to open a media frame for a set of flows for a very specific block (e.g., image states, gallery states, audio states, playlist states, featured image states etc.). We either need an API that determines what set of states get loaded, or we need to pass those states to the component explicitly (or create a set of components that wrap `MediaUpload` for specific states).

I agree with you that having individual props for different states like "gallery" isn't the most clean, but I'm not sure what approach you would like to see taken.

Reply:
As I said, I'm ok with this approach for now. I think at some point, I'd like some work on the media library to be more flexible (just using the same controller with options), but obviously that just my opinion and even if shared, not something for 5.0